src/jaxsim/high_level/model.py

```diff
@@ -1479,6 +1490,88 @@ def integrate(
             },
         )
 
+    # ==============
+    # Motor dynamics
+    # ==============
+
+    @functools.partial(oop.jax_tf.method_rw, static_argnames=["joint_names"])
+    def set_motor_inertias(
+        self, inertias: jtp.Vector, joint_names: tuple[str, ...] = None
+    ) -> None:
+        joint_names = joint_names or self.joint_names()
+
+        if inertias.size != len(joint_names):
+            raise ValueError("Wrong arguments size", inertias.size, len(joint_names))
+
+        self.physics_model._joint_motor_inertia.update(
+            dict(zip(self.physics_model._joint_motor_inertia, inertias))
+        )
+
+        logging.info("Setting attribute `motor_inertias`")
+
+    @functools.partial(oop.jax_tf.method_rw, jit=False)
+    def set_motor_gear_ratios(
+        self, gear_ratios: jtp.Vector, joint_names: tuple[str, ...] = None
+    ) -> None:
+        joint_names = joint_names or self.joint_names()
+
+        if gear_ratios.size != len(joint_names):
+            raise ValueError("Wrong arguments size", gear_ratios.size, len(joint_names))
+
+        # Check on gear ratios if motor_inertias are not zero
+        for idx, gr in enumerate(gear_ratios):
+            if gr != 0 and self.motor_inertias()[idx] == 0:
+                raise ValueError(
+                    f"Zero motor inertia with non-zero gear ratio found in position {idx}"
+                )
+
+        self.physics_model._joint_motor_gear_ratio.update(
+            dict(zip(self.physics_model._joint_motor_gear_ratio, gear_ratios))
+        )
+
+        logging.info("Setting attribute `motor_gear_ratios`")
+
+    @functools.partial(oop.jax_tf.method_rw, static_argnames=["joint_names"])
+    def set_motor_viscous_frictions(
+        self, viscous_frictions: jtp.Vector, joint_names: tuple[str, ...] = None
+    ) -> None:
+        joint_names = joint_names or self.joint_names()
+
+        if viscous_frictions.size != len(joint_names):
+            raise ValueError(
+                "Wrong arguments size", viscous_frictions.size, len(joint_names)
+            )
+
+        self.physics_model._joint_motor_viscous_friction.update(
+            dict(
+                zip(
+                    self.physics_model._joint_motor_viscous_friction,
+                    viscous_frictions,
+                )
+            )
+        )
+
+        # self.physics_model._joint_motor_viscous_friction |= dict(
+        #     zip(
+        #         self.physics_model._joint_motor_viscous_friction.keys(),
+        #         viscous_frictions,
+        #     )
+        # )
+
+        logging.info("Setting attribute `motor_viscous_frictions`")
+
+    @functools.partial(oop.jax_tf.method_ro, jit=False)
+    def motor_inertias(self) -> jtp.Vector:
+        return jnp.array([*self.physics_model._joint_motor_inertia.values()])
```

Comment:
let's enforce `dtype=float` also here

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/62/commits/9d4754b5a054c42f4248348ea1af20d1079fbb6f